### ISSUE TITLE
Names often overun boxes in browser output

### ISSUE THREAD

**Issue by Neriderc** [Neriderc/GVExport#37]
I find this issue on many (perhaps most) of the boxes in the browser rendering. This is not an issue for server side generation, which to me says that the issue is in viz.js since it's using the same DOT file to generate the output. 

![image](https://user-images.githubusercontent.com/89974636/174040665-6989a56e-299b-4f82-889f-fb592bb791cb.png)

I honestly am not sure how to start going about fixing this, as the most [recent version of viz.js I can find ](https://github.com/aduh95/viz.js) has the same issue when testing on their [demo site](http://aduh95.github.io/viz.js-playground/).

This virtually brings the work in #23 to a standstill, as the output is not the same as server-side generation.

Setting the width of the boxes in the HTML tags may work... a bit of a hack because the tags seem to be stripped when the DOT is generated so it would need to be added after generation, but I confirmed on the demo site that adding width="150" in the TD tags of the DOT file *does* make the boxes bigger, it's just a bit messy.

Any other ideas appreciated!

**Comment by Neriderc** [comment]
Is anyone able to confirm they get this bug as well? For me this happens 100% of the time, in every diagram type, and seems to happen regardless of other settings. However, I've looked back at when others have posted screen shots and they don't seem to have this issue.

So is there anyone else that is seeing this issue?
![image](https://user-images.githubusercontent.com/89974636/178846144-eb308d6b-65ec-4e9a-858c-9b4dbf37cd75.png)

**Comment by hartenthaler** [comment]
I have never seen this problem.

**Comment by Neriderc** [comment]
I thought a bit about this, and thought the font could be the issue. The default font is set to "Sans" but this could be different on different OSs. I found setting this to "Arial" fixed the issue.

What operating system are you using (both browser and server)? Does it work well with different browsers? Do you get the same output in the browser as when you generate server side, or are there differences?

**Comment by schuco** [comment]
In my case (old version for wt 2.0) I observed this in only very very few cases. It would not need fixing this.
It seems I have Times New Roman fonts client side and Arial serverside.
In the original version of GVExport (before migrated to wt2.0) there were a lot more user-options. Among others you could choose font, font size, but also e.g. colours for male, female and non related persons.

**Comment by Neriderc** [comment]
> In my case (old version for wt 2.0) I observed this in only very very few cases. It would not need fixing this. It seems I have Times New Roman fonts client side and Arial serverside. In the original version of GVExport (before migrated to wt2.0) there were a lot more user-options. Among others you could choose font, font size, but also e.g. colours for male, female and non related persons.

Thanks! I tested Times New Roman and I don't have any issues with that either!

**Comment by Neriderc** [comment]
Whoops accidentally closed it. There's something odd here that's specific to my setup, and it only seems to affect the browser render not the server side one even in my development instance that is running both on the same machine. I'm gonna leave this open while I investigate (in particular I want to test some of my other devices once I get a chance), but I think the major thing here is that this most likely is not the barrier to client-side rendering that I thought it was. I think I'll add client-side rendering back to my list of things to investigate further as I put that on hold based on this issue (as the client-side output will be identical to the browser render).

**Comment by hartenthaler** [comment]
- I have just tested using Firefox and Edge on Windows and cannot see any problem with the sizes of the boxes.
- testing using Edge and Safari on iOS was strange: I saw the problem in both browsers, but during playing around it disappeared and I was not able to bring it up again.

**Comment by Neriderc** [comment]
Thanks for testing! My laptop is running Fedora Linux, which is where my development environment is installed. My production server runs Ubuntu Linux and I see the same thing there, but I'm only testing by accessing from a browser on my laptop as the server doesn't have a GUI to run a browser in.

I'm away from home but will test on Windows when I get home so I can confirm if it works ok there. I'd guess it's probably related to the default "sans" font on Fedora, though that doesn't explain why the server-side render works fine.

**Comment by Neriderc** [comment]
Right, thanks for your help guys! I have now tested on Windows and don't see this issue at all.

I also tested out some other operating systems running in VMs. Every version of Linux I tried _did_ have the issue, even though they all use different default fonts. Setting the font to one of the Windows fonts (like Arial, Times New Roman) seems to solve the issue.

Because this is environmental, I think I'll just add a section to the Troubleshooting page on the wiki to say it's a known issue and that changing the font can help. I just don't see this being a priority to spend any more time on.
